### PR TITLE
fix some pylint issues

### PR DESCRIPTION
**Pull request by guotuofeng**
## Describe your changes
fix some pylint issues
## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Format your code by running `pre-commit run --all-files`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: olive/data/component/dataset.py [hunk 1]
```diff
@@ -80,8 +80,8 @@ def to_hf_dataset(self, label_name="label"):
             data_dict = {k: [] for k in first_input}
             data_dict[label_name] = []
             # loop over the dataset
-            for i in range(len(self)):  # pylint: disable=consider-using-enumerate
-                data, label = deepcopy(self[i])
+            for _, d in enumerate(self):
+                data, label = deepcopy(d)
```
Comment (justinchuby) [inline olive/data/component/dataset.py:84]: simply do `for d in self`?

File: test/multiple_ep/test_python_env_system.py [hunk 2]
```diff
@@ -50,15 +48,14 @@ def test_run_pass_evaluate_windows(self):
         assert openvino_res[tuple(engine.pass_flows[0])]["metrics"]["latency-avg"]
 
     @pytest.mark.skipif(platform.system() == "Windows", reason="Test for Linux only")
-    def test_run_pass_evaluate_linux(self):
+    def test_run_pass_evaluate_linux(self, tmpdir):
```
Comment (justinchuby) [inline test/multiple_ep/test_python_env_system.py:51]: Curious how tmpdir is provided?

Comment (guotuofeng) [reply test/multiple_ep/test_python_env_system.py:51]: tmpdir is pytest fixture

Comment (justinchuby) [reply test/multiple_ep/test_python_env_system.py:51]: That's neat!

**Pull request merged**